### PR TITLE
Add unit test for OpenTelemetry plugin

### PR DESCRIPTION
**Pull request by JooKS-me**
Refer #11102 .

### CODE REVIEW

**Comment by JooKS-me** [comment]
@yu199195 please have a review~

File: shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-tracing/shardingsphere-agent-tracing-opentelemetry/src/test/java/org/apache/shardingsphere/agent/plugin/tracing/opentelemetry/service/OpenTelemetryTracingPluginBootServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.junit.After;
+import org.junit.Test;
+
+import java.util.Properties;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.hamcrest.CoreMatchers.is;
+
+public class OpenTelemetryTracingPluginBootServiceTest {
```
Comment (tristaZero) [inline shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-tracing/shardingsphere-agent-tracing-opentelemetry/src/test/java/org/apache/shardingsphere/agent/plugin/tracing/opentelemetry/service/OpenTelemetryTracingPluginBootServiceTest.java:31]: Hi all the unit tests are suggested to be a final class, as it is not likely to extend them, right?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-tracing/shardingsphere-agent-tracing-opentelemetry/src/test/java/org/apache/shardingsphere/agent/plugin/tracing/opentelemetry/service/OpenTelemetryTracingPluginBootServiceTest.java (hunk 1 above)
Comment (JooKS-me) [reply shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-tracing/shardingsphere-agent-tracing-opentelemetry/src/test/java/org/apache/shardingsphere/agent/plugin/tracing/opentelemetry/service/OpenTelemetryTracingPluginBootServiceTest.java:31]: Thanks for pointing out, I will modify this.

Comment (JooKS-me) [reply shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-tracing/shardingsphere-agent-tracing-opentelemetry/src/test/java/org/apache/shardingsphere/agent/plugin/tracing/opentelemetry/service/OpenTelemetryTracingPluginBootServiceTest.java]: @tristaZero hi, I have changed it~

File: shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-tracing/shardingsphere-agent-tracing-opentelemetry/src/test/java/org/apache/shardingsphere/agent/plugin/tracing/opentelemetry/collector/OpenTelemetryCollector.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import io.opentelemetry.sdk.testing.exporter.InMemorySpanExporter;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import org.apache.shardingsphere.agent.plugin.tracing.rule.CollectorRule;
+import org.junit.rules.ExternalResource;
+
+import java.util.List;
+
+public class OpenTelemetryCollector extends ExternalResource implements CollectorRule {
```
Comment (tristaZero) [inline shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-tracing/shardingsphere-agent-tracing-opentelemetry/src/test/java/org/apache/shardingsphere/agent/plugin/tracing/opentelemetry/collector/OpenTelemetryCollector.java:31]: Is it possible some classes will extend this one? Or define it as a final class?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-tracing/shardingsphere-agent-tracing-opentelemetry/src/test/java/org/apache/shardingsphere/agent/plugin/tracing/opentelemetry/collector/OpenTelemetryCollector.java (hunk 2 above)
Comment (JooKS-me) [reply shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-tracing/shardingsphere-agent-tracing-opentelemetry/src/test/java/org/apache/shardingsphere/agent/plugin/tracing/opentelemetry/collector/OpenTelemetryCollector.java]: Sry, I missed it...

**Review by tristaZero — APPROVED** [review]
@JooKS-me Nice.

**Comment by tristaZero** [comment]
Sorry, this PR still needs @yu199195 's approval. Do you think there is any improvement for this PR?

**Pull request merged**